### DISCUSSION TITLE
RAM screen buffer?

### DISCUSSION THREAD

**Discussion by quarterturn** [Bodmer/TFT_eSPI#1202] [Q&A - General]
I'd like to do a 3D library which will naturally draw filled triangles to the screen, but this is typically too slow over SPI to do on a per poly basis if you want to do animations. Does your library support blitting from RAM to the screen? I see there's a DMA example using flash. Any ideas for adapting it to RAM?

**Comment by Bodmer** [comment]
You will need a processor with a lot of RAM. The[ example here](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/DMA%20test/Bouncy_Circles) uses Sprites and DMA to update the screen, it uses 153K RAM, runs on an RP2040 and may run on an ESP32.

**Reply by handmade0octopus** [reply]
https://www.reddit.com/r/esp32/comments/nuqpvw/tft_espi_library_is_awesome_dme_320x240_ili9341/ Confirming it runs nicely on ESP32 :)

On side note, is there a way to use something like tft.println() and .setCursor() but to write to buffer like this? I am thinking of making some basic library and this would really help as it seems to be better idea to just do refresh like this.

**Comment by quarterturn** [comment]
On Thu, May 27, 2021, at 6:55 PM, Bodmer wrote:
> 
> 
> You will need a processor with a lot of RAM. The example here 
> <https://github.com/Bodmer/TFT_eSPI/tree/master/examples/DMA%20test/Bouncy_Circles> uses Sprites and DMA to update the screen, it uses 153K RAM, runs on an RP2040 and may run on an ESP32.
> 

Interesting. If a sprite can be the whole screen this may work for me. I'm targeting a 160x128 
ST7335 and an ATSAMD51 so there should be enough RAM for double-buffering at that size. If I get something working I'll link to my github repo in case you'd like to include it as an example.

Thanks!
